go/vt/vtgate/planbuilder/testdata/reference_cases.json

```diff
@@ -22,7 +22,7 @@
       "Original": "select * from ambiguous_ref_with_source",
       "Instructions": {
         "OperatorType": "Route",
-        "Variant": "Reference",
+        "Variant": "Unsharded",
```

Comment:
`ambiguous_ref_with_source` exists both as an unsharded table in the `main` keyspace, and as a reference table in the `user` keyspace. The latter is a copy of the unsharded table spread out to all shards so that all joins can be local.

During route planning we have decided that we want to send the query to the unsharded `main` keyspace. The OpCode is more accurate if it's `Unsharded` for this route.